tests/apitest/src/CoreServices/FSEventStream.cs

```diff
@@ -0,0 +1,169 @@
+		ulong st_dev;
+		FSEventStream fsEventStream;
+		NSArray pathsToWatchRelativeToDevice;
+		EventWaitHandle waitHandle;
+		AutoResetEvent watchEventCalled;
+		AutoResetEvent excludeEventCalled;
+
+		[TestFixtureSetUp]
+		public void Init ()
+		{
+			dir = Directory.CreateDirectory ("FSEventStreamTests");
+			fileToWatch = File.Create ("FSEventStreamTests/TempFileToWatch.txt");
+			fileToExclude = File.Create ("FSEventStreamTests/TempFileToExclude.txt");
+			var process = new Process () {
+				StartInfo = new ProcessStartInfo {
+					FileName = "/usr/bin/stat",
+					Arguments = $"-f '%d' FSEventStreamTests/TempFileToWatch.txt",
+					RedirectStandardOutput = true,
+					UseShellExecute = false,
+					CreateNoWindow = true,
+				}
+			};
+			process.Start ();
+			string result = process.StandardOutput.ReadToEnd ();
+			process.WaitForExit ();
+			st_dev = UInt64.Parse (result);
+			pathsToWatchRelativeToDevice = NSArray.FromStrings (new [] { "FSEventStreamTests" });
+			fileToWatch.Close ();
+			fileToExclude.Close ();
+		}
+
+		[TestFixtureTearDown]
+		public void Destroy ()
+		{
+			Directory.Delete ("FSEventStreamTests");
+			pathsToWatchRelativeToDevice.Dispose ();
+		}
+
+		[SetUp]
+		public void CreateFSEventStream ()
+		{
+			watchedFileChanged = false;
+			exludedFileChanged = false;
+			fsEventStream = new FSEventStream (st_dev, pathsToWatchRelativeToDevice, ulong.MaxValue, TimeSpan.MinValue, FSEventStreamCreateFlags.None);
+			Assert.IsNotNull (fsEventStream, "Null");
+			fsEventStream.Events += FsEventStream_WatchEvents;
+			fsEventStream.ScheduleWithRunLoop (CFRunLoop.Current);
+			waitHandle = new EventWaitHandle (false, EventResetMode.AutoReset);
+			watchEventCalled = new AutoResetEvent (false);
+			excludeEventCalled = new AutoResetEvent (false);
+			fsEventStream.Start ();
+		}
+
+		public void FsEventStream_WatchEvents (object sender, FSEventStreamEventsArgs args)
+		{
+			Console.WriteLine ("FsEventStream_WatchEvents");
+			watchedFileChanged = true;
+			//watchEventCalled.Set ();
+		}
+
+		[TearDown]
+		public void EndFSEventStream ()
+		{
+			fsEventStream.Stop ();
+			fsEventStream.Invalidate ();
+			fsEventStream.Dispose ();
+		}
+
+		[Test]
+		public void GetDeviceBeingWatchedTest ()
+		{
+			var deviceId = fsEventStream.GetDevice ();
+			Assert.AreEqual (st_dev, deviceId, "Device ID"); //Check if device id is the same
+		}
+
+		[Test]
+		public void FSEventFileChangedTest ()
+		{
+			string path = "FSEventStreamTests/TempFileToWatch.txt";
+			Assert.IsTrue (File.Exists (path));;
+
+			File.AppendAllText (path, "Hello World!");
+			//watchEventCalled.WaitOne (Int32.MaxValue);
+			//waitHandle.WaitOne ();
+			Assert.IsTrue (watchedFileChanged);
+		}
+
+		//[Test]
+		public void UnscheduleFromRunLoopTest ()
+		{
+			fsEventStream.UnscheduleFromRunLoop (CFRunLoop.Current);
+			File.AppendAllText ("FSEventStreamTests/TempFileToWatch.txt", "Hello World!");
+			watchEventCalled.WaitOne (Int32.MaxValue);
```

Comment:
The same here: you must call TestRuntime.RunAsync (but with a short timeout, since it will take that long every time).